### PR TITLE
GUVNOR-2845: Data modeller: Add new Java 8 Local Date/Time types

### PR DESCRIPTION
**Pull request by mcimbora**

### CODE REVIEW

**Comment by mcimbora** [comment]
Add XStream Converter & JAXB XmlAdapter definitions for java.time.LocalDate, java.time.LocalDateTime, java.time.LocalTime.

Part of:
https://github.com/droolsjbpm/droolsjbpm-knowledge/pull/198
https://github.com/droolsjbpm/droolsjbpm-integration/pull/789
https://github.com/droolsjbpm/kie-wb-common/pull/631

@ge0ffrey Can you please have a look?

File: kie-internal/src/main/java/org/kie/internal/xstream/LocalDateXStreamConverter.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+import java.time.LocalDate;
+
+import com.thoughtworks.xstream.converters.Converter;
+import com.thoughtworks.xstream.converters.MarshallingContext;
+import com.thoughtworks.xstream.converters.UnmarshallingContext;
+import com.thoughtworks.xstream.io.HierarchicalStreamReader;
+import com.thoughtworks.xstream.io.HierarchicalStreamWriter;
+
+public class LocalDateXStreamConverter implements Converter {
```
Comment (ge0ffrey) [inline kie-internal/src/main/java/org/kie/internal/xstream/LocalDateXStreamConverter.java]: This belongs in XStream, let's see what this discussion teaches us:
  https://groups.google.com/forum/#!topic/xstream-user/IXAkhHVdOGU

I suspect it doesn't support it yet (surprising to me!), so we'll need to accept this Converter here temporarily, but create an issue on XStream, add a comment here like "temporarly workaroudn until ...xstreamIssueUrl... is fixed" and as soon as XStream support it, we upgrade the xstream dependency and remove this class.

Comment (mcimbora) [inline kie-internal/src/main/java/org/kie/internal/xstream/LocalDateXStreamConverter.java]: @ge0ffrey +1

**Comment by mcimbora** [comment]
@ge0ffrey PR updated.
- Fix the output format to match XSD date/time types (http://www.w3schools.com/xml/schema_dtypes_date.asp)
- Add OffsetDateTime converter/xml adapter

**Pull request merged**